editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.tsx

```diff
@@ -124,47 +129,49 @@ function Component(props: Props) {
       isValid={slots.every((slot) => slot.url && slot.status === "success")}
     >
       <QuestionHeader {...props} />
-        <DropzoneContainer>
-          <FileStatus status={fileUploadStatus} />
-          <ErrorWrapper error={validationError} id={props.id}>
+      <DropzoneContainer>
+        <FileStatus status={fileUploadStatus} />
+        <ErrorWrapper error={validationError} id={props.id}>
           <Dropzone
             slots={slots}
             setSlots={setSlots}
             setFileUploadStatus={setFileUploadStatus}
           />
-          </ErrorWrapper>
-          <Box>
-            <Typography fontWeight={FONT_WEIGHT_BOLD}>
-              Required files
-            </Typography>
-            {fileList.required.map((fileType) => (
-              <InteractiveFileListItem
-                name={fileType.key}
-                moreInformation={fileType.moreInformation}
-              />
-            ))}
-          </Box>
-        </DropzoneContainer>
-        {Boolean(slots.length) && (
-          <Typography mb={2} fontWeight={FONT_WEIGHT_BOLD}>
-            Your uploaded files
-          </Typography>
-        )}
-        {slots.map((slot) => {
-          return (
-            <UploadedFileCard
-              {...slot}
-              key={slot.id}
-              tags={["Test1", "Test2", "Test3"]}
-              removeFile={() => {
-                setSlots(
-                  slots.filter((currentSlot) => currentSlot.file !== slot.file)
-                );
-                setFileUploadStatus(`${slot.file.path} was deleted`);
-              }}
+        </ErrorWrapper>
+        <Box>
+          <Typography fontWeight={FONT_WEIGHT_BOLD}>Required files</Typography>
+          {fileList.required.map((fileType) => (
+            <InteractiveFileListItem
+              name={fileType.key}
+              moreInformation={fileType.moreInformation}
             />
-          );
-        })}
+          ))}
+        </Box>
+      </DropzoneContainer>
+      {Boolean(slots.length) && (
+        <Typography mb={2} fontWeight={FONT_WEIGHT_BOLD}>
+          Your uploaded files
+        </Typography>
+      )}
+      {showModal && (
+        <FileTaggingModal uploadedFiles={slots} setShowModal={setShowModal} />
+      )}
+      {slots.map((slot) => {
+        return (
+          <UploadedFileCard
+            {...slot}
+            key={slot.id}
+            tags={["Test1", "Test2", "Test3"]}
+            onChange={() => setShowModal(true)}
+            removeFile={() => {
+              setSlots(
+                slots.filter((currentSlot) => currentSlot.file !== slot.file)
+              );
+              setFileUploadStatus(`${slot.file.path} was deleted`);
+            }}
+          />
+        );
+      })}
```

Comment:
these are mostly indentation corrections - sorry for noisy diff! 

the juicy bit is the new `FileTaggingModal` component below